sdk/PowerBI.Core/Security/GatewayCredentials.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerBI.Core.Cryptography;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.PowerBI.Core.Security
+{
+    /// <summary>
+    /// The gateway credentials used to authticate to your datasources
+    /// </summary>
+    public class GatewayCredentials
+    {
+        /// <summary>
+        /// The username used to authorize to the gateway
```

Comment:
Maybe we should GatewayCredentials be just the username and password. Then we have a static class that encrypts and decrypts the GatewayCredentials. I prefer the decoupling.